eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -0,0 +1,224 @@
+        $query->select('id', 'current_version')
+            ->from('ezcontentobject')
+            ->where($query->expr->eq('status', ContentInfo::STATUS_PUBLISHED));
+
+        $stmt = $query->prepare();
+        $stmt->execute();
+
+        $this->searchHandler->purgeIndex();
+
+        $this->output->writeln('Indexing Content...');
+
+        /** @var \Symfony\Component\Console\Helper\ProgressBar $progress */
+        $progress = new ProgressBar($this->output);
+        $progress->start($totalCount);
+        $i = 0;
+        do {
+            $contentObjects = [];
+
+            for ($k = 0; $k <= $bulkCount; ++$k) {
+                if (!$row = $stmt->fetch(PDO::FETCH_ASSOC)) {
+                    break;
+                }
+
+                $contentObjects[] = $this->persistenceHandler->contentHandler()->load(
+                    $row['id'],
+                    $row['current_version']
+                );
+            }
+
+            foreach ($contentObjects as $contentObject) {
+                $this->searchHandler->indexContent($contentObject);
+            }
+
+            $progress->advance($k);
+        } while (($i += $bulkCount) < $totalCount);
+
+        $progress->finish();
+    }
+
+    /**
+     * Wrapper for indexing locations.
+     *
+     * @param int $bulkCount a number of node rows to fetch in a single batch
+     */
+    private function createLocationsIndex($bulkCount)
+    {
+        $query = $this->databaseHandler->createSelectQuery();
+        $query
+            ->select('count(node_id)')
+            ->from('ezcontentobject_tree')
+            ->where(
+                $query->expr->neq(
+                    $this->databaseHandler->quoteColumn('contentobject_id'),
+                    $query->bindValue(0, null, PDO::PARAM_INT)
+                )
+            );
+        $stmt = $query->prepare();
+        $stmt->execute();
+        $totalCount = $stmt->fetchColumn();
+
+        $query = $this->databaseHandler->createSelectQuery();
+        $query
+            ->select('node_id')
+            ->from('ezcontentobject_tree')
+            ->where(
+                $query->expr->neq(
+                    $this->databaseHandler->quoteColumn('contentobject_id'),
+                    $query->bindValue(0, null, PDO::PARAM_INT)
+                )
+            );
+
+        $stmt = $query->prepare();
+        $stmt->execute();
+
+        $this->output->writeln('Indexing Locations...');
+
+        /** @var \Symfony\Component\Console\Helper\ProgressBar $progress */
+        $progress = new ProgressBar($this->output);
+        $progress->start($totalCount);
+        $i = 0;
+        do {
+            $locations = [];
+
+            for ($k = 0; $k <= $bulkCount; ++$k) {
+                if (!$row = $stmt->fetch(PDO::FETCH_ASSOC)) {
+                    break;
+                }
+
+                $locations[] = $this->persistenceHandler->locationHandler()->load($row['node_id']);
```

Comment:
Added exception handling in eee88f4 to keep code consistent, but haven't seen a case when an error occurs here.